master/buildbot/test/unit/test_worker_base.py

```diff
@@ -471,6 +471,37 @@ def test_shutdownRequested(self):
         self.assertEqual(worker.worker_status.getGraceful(), True)
 
 
+class TestAbstractLatentWorker(unittest.SynchronousTestCase):
+
+    def setUp(self):
+        self.master = fakemaster.make_master(wantDb=True, wantData=True,
+                                             testcase=self)
+        self.botmaster = self.master.botmaster
+        self.master.workers.disownServiceParent()
+        self.workers = self.master.workers = bworkermanager.FakeWorkerManager()
+        self.workers.setServiceParent(self.master)
+        self.clock = task.Clock()
+        self.patch(reactor, 'callLater', self.clock.callLater)
+        self.patch(reactor, 'seconds', self.clock.seconds)
+
+    def do_test_reconfigService(self, old, new, existingRegistration=True):
+        old.parent = self.master
+        if existingRegistration:
+            old.registration = bworkermanager.FakeWorkerRegistration(old)
+        old.missing_timer = mock.Mock(name='missing_timer')
+        self.successResultOf(old.startService())
```

Comment:
I wasn't aware of this method.

This fails with twisted 12:

  File "/home/travis/build/buildbot/buildbot/master/buildbot/test/unit/test_worker_base.py", line 500, in test_reconfigService
    self.do_test_reconfigService(old, new)
  File "/home/travis/build/buildbot/buildbot/master/buildbot/test/unit/test_worker_base.py", line 492, in do_test_reconfigService
    self.successResultOf(old.startService())
exceptions.AttributeError: 'TestAbstractLatentWorker' object has no attribute 'successResultOf'

Looks like was introduced in 13.
I am not against dropping such old version of twisted, but we should probably ask the ml first.